### PR TITLE
feat: bump Firebase Android SDK to `29.3.0`

### LINKED ISSUE #6749: 🐛 [cloud_firestore] After some time, Cloud Firestore sync stops working
## Bug report

Both on Android and iOS, the sync works fine when the application starts and keeps working for some time. However, after some time (not sure how much) sync stops working - sending doesn't work, and receiving in another instance of the app doesn't receive changes (I can see this by changing data directly in the console). When the application is force-stopped and restarted, it starts working again immediately. This behavior is 100% consistent between Android and iOS.

I started having the issue only recently, maybe 2-3 weeks ago. Before this, this application and Firestore integration had been working flawlessly for months and there have been no code changes at all since May 19, and no code changes to anything related to Firebase or Firestore for half a year (except for version updates).

Has anything changed in your backend, and is some change in code required for it to start working without issues again?

I upgraded everything about a week ago but it didn't help.

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, 2.2.3, on macOS 11.5.1 20G80 darwin-x64, locale en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[!] Android Studio (version 2020.3)
    ✗ Unable to find bundled Java version.
[✓] IntelliJ IDEA Ultimate Edition (version 2021.2)
[✓] VS Code (version 1.58.2)
[✓] Connected device (2 available)

! Doctor found issues in 1 category.
```
</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.13.4
Flutter SDK 2.2.3
mu_project 1.2.0-dev

dependencies:
- blocs 0.0.0 [bloc equatable flutter google_sign_in meta model repository rxdart]
- cloud_firestore 2.4.0 [cloud_firestore_platform_interface cloud_firestore_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_crashlytics 2.1.1 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_bloc 7.0.1 [flutter bloc provider]
- flutter_localizations 0.0.0 [flutter intl characters clock collection meta path typed_data vector_math]
- flutter_slidable 0.6.0 [flutter]
- flutter_svg 0.22.0 [flutter meta path_drawing vector_math xml]
- googleapis 4.0.0 [_discoveryapis_commons http]
- http 0.13.3 [async http_parser meta path pedantic]
- hydrated_bloc 7.0.1 [bloc hive meta synchronized]
- intl 0.17.0 [clock path]
- json_annotation 4.0.1
- meta 1.3.0
- model 0.0.0 [equatable meta]
- package_info_plus 1.0.4 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- path_provider 2.0.2 [flutter path_provider_platform_interface path_provider_macos path_provider_linux path_provider_windows]
- provider 5.0.0 [collection flutter nested]
- repository 0.0.0 [cloud_firestore firebase_auth flutter google_sign_in googleapis http json_annotation meta rxdart]
- wakelock 0.5.2 [flutter meta wakelock_macos wakelock_platform_interface wakelock_web wakelock_windows]

dev dependencies:
- build_runner 2.0.6 [args async analyzer build build_config build_daemon build_resolvers build_runner_core code_builder collection crypto dart_style frontend_server_client glob graphs http_multi_server io js logging meta mime package_config path pedantic pool pub_semver pubspec_parse shelf shelf_web_socket stack_trace stream_transform timing watcher web_socket_channel yaml]
- flutter_launcher_icons 0.9.1 [args image path yaml]
- json_serializable 4.1.4 [analyzer build build_config collection json_annotation meta path source_gen]

transitive dependencies:
- _discoveryapis_commons 1.0.1 [http http_parser meta]
- _fe_analyzer_shared 22.0.0 [meta]
- analyzer 1.7.1 [_fe_analyzer_shared cli_util collection convert crypto glob meta package_config path pub_semver source_span watcher yaml pedantic]
- archive 3.1.2 [crypto path]
- args 2.2.0
- async 2.8.1 [collection meta]
- bloc 7.0.0 [meta]
- build 2.0.3 [analyzer async convert crypto glob logging meta path]
- build_config 1.0.0 [checked_yaml json_annotation path pubspec_parse yaml]
- build_daemon 3.0.0 [built_collection built_value http_multi_server logging pedantic path pool shelf shelf_web_socket stream_transform watcher web_socket_channel]
- build_resolvers 2.0.4 [analyzer async build crypto graphs logging path package_config pool pub_semver stream_transform yaml]
- build_runner_core 7.0.1 [async build build_config build_resolvers collection convert crypto glob graphs json_annotation logging meta path package_config pedantic pool timing watcher yaml]
- built_collection 5.1.0
- built_value 8.1.1 [built_collection collection fixnum meta]
- characters 1.1.0
- charcode 1.3.1
- checked_yaml 2.0.1 [json_annotation source_span yaml]
- cli_util 0.3.3 [meta path]
- clock 1.1.0
- cloud_firestore_platform_interface 5.3.0 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 2.3.0 [cloud_firestore_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- code_builder 4.1.0 [built_collection built_value collection matcher meta]
- collection 1.15.0
- convert 3.0.1 [typed_data]
- crypto 3.0.1 [collection typed_data]
- dart_style 2.0.3 [analyzer args path pub_semver source_span]
- equatable 2.0.3 [collection meta]
- ffi 1.1.2
- file 6.1.2 [meta path]
- firebase_auth 3.0.1 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_auth_platform_interface 6.0.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 3.0.0 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core 1.4.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_core_platform_interface 4.0.1 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.1.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_crashlytics_platform_interface 3.1.0 [collection firebase_core flutter meta plugin_platform_interface]
- fixnum 1.0.0
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- frontend_server_client 2.1.0 [async path]
- glob 2.0.1 [async collection file path pedantic string_scanner]
- google_sign_in 5.0.5 [flutter google_sign_in_platform_interface google_sign_in_web meta]
- google_sign_in_platform_interface 2.0.1 [flutter meta quiver]
- google_sign_in_web 0.10.0 [google_sign_in_platform_interface flutter flutter_web_plugins meta js]
- graphs 2.0.0
- hive 2.0.4 [meta crypto]
- http_multi_server 3.0.1 [async]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image 3.0.2 [archive meta xml]
- io 1.0.3 [meta path string_scanner]
- js 0.6.3
- logging 1.0.1
- matcher 0.12.10 [stack_trace]
- mime 1.0.0
- nested 1.0.0 [flutter]
- package_config 2.0.0 [path]
- package_info_plus_linux 1.0.3 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.1.1 [flutter]
- package_info_plus_platform_interface 1.0.2 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.3 [flutter flutter_web_plugins http meta package_info_plus_platform_interface]
- package_info_plus_windows 1.0.3 [package_info_plus_platform_interface ffi flutter win32]
- path 1.8.0
- path_drawing 0.5.1 [vector_math meta path_parsing flutter]
- path_parsing 0.2.1 [vector_math meta]
- path_provider_linux 2.0.0 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 2.0.0 [flutter]
- path_provider_platform_interface 2.0.1 [flutter meta platform plugin_platform_interface]
- path_provider_windows 2.0.1 [path_provider_platform_interface meta path flutter ffi win32]
- pedantic 1.11.1
- petitparser 4.1.0 [meta]
- platform 3.0.0
- plugin_platform_interface 2.0.1 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.2.3 [file path platform]
- pub_semver 2.0.0 [collection]
- pubspec_parse 1.0.0 [checked_yaml collection json_annotation pub_semver yaml]
- quiver 3.0.1 [matcher]
- rxdart 0.27.1
- shelf 1.2.0 [async collection http_parser path stack_trace stream_channel]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_gen 1.0.3 [analyzer async build dart_style glob meta path pedantic source_span yaml]
- source_span 1.8.1 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- stream_transform 2.0.0
- string_scanner 1.1.0 [charcode source_span]
- synchronized 3.0.0
- term_glyph 1.2.0
- timing 1.0.0 [json_annotation]
- typed_data 1.3.0 [collection]
- vector_math 2.1.0
- wakelock_macos 0.1.0+1 [flutter flutter_web_plugins wakelock_platform_interface]
- wakelock_platform_interface 0.2.1+1 [flutter meta]
- wakelock_web 0.2.0+1 [flutter flutter_web_plugins js wakelock_platform_interface]
- wakelock_windows 0.1.0 [flutter wakelock_platform_interface win32]
- watcher 1.0.0 [async path pedantic]
- web_socket_channel 2.1.0 [async crypto stream_channel]
- win32 2.2.5 [ffi]
- xdg_directories 0.2.0 [meta path process]
- xml 5.1.2 [collection meta petitparser]
- yaml 3.1.0 [collection source_span string_scanner]
```

</details>

---

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

Bump android sdk versions.

## Related Issues

fixes https://github.com/FirebaseExtended/flutterfire/issues/7988
fixes #6749 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by russellwheatley** [comment]
We're going to wait until Firebase release a fix for [this](https://github.com/firebase/firebase-android-sdk/issues/3526) which has a PR already approved [here](https://github.com/firebase/firebase-android-sdk/pull/3533) & I believe will be released soonish. Otherwise, we will fix the Firestore bug and replace it with a failing Firebase Messaging build issue.

**Comment by larssn** [comment]
This will close https://github.com/FirebaseExtended/flutterfire/issues/6749

File: packages/firebase_core/firebase_core/android/gradle.properties [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # https://firebase.google.com/support/release-notes/android
-FirebaseSDKVersion=29.0.4
+FirebaseSDKVersion=29.2.0
```
Comment (Salakar) [inline packages/firebase_core/firebase_core/android/gradle.properties:2]:
```suggestion
FirebaseSDKVersion=29.2.1
```

File: docs/manual-installation.mdx [hunk 2]
```diff
@@ -109,7 +109,7 @@ In the `/android/app/build.gradle` file, you can provide your own versions using
 ```groovy
 rootProject.ext {
   set('FlutterFire', [
-    FirebaseSDKVersion: '29.0.4'
+    FirebaseSDKVersion: '29.2.0'
```
Comment (Salakar) [inline docs/manual-installation.mdx:112]:
```suggestion
    FirebaseSDKVersion: '29.2.1'
```

**Review by larssn — APPROVED** [review]

**Comment by bswhite1** [comment]
Any timeline on when this will get merged and released?

**Review by larssn — APPROVED** [review]

**Comment by larssn** [comment]
@russellwheatley Could you update the op with a "Fixes #6749"? 🙂

**Review by Salakar — APPROVED** [review]

**Review by rrousselGit — APPROVED** [review]

**Pull request merged**